python/hsfs/core/training_dataset_engine.py

```diff
@@ -107,3 +113,81 @@ def delete_tag(self, training_dataset, name):
     def get_tags(self, training_dataset, name):
         """Get tag with a certain name or all tags for a training dataset."""
         return [tag.to_dict() for tag in self._tags_api.get(training_dataset, name)]
+
+    def get_serving_vector(self, training_dataset, entry):
+        """Get ... for a training dataset."""
+
+        serving_vector = []
+
+        if training_dataset.prepared_statements is None:
+            self._init_prepared_statement(training_dataset)
+
+        prepared_statements = training_dataset.prepared_statements
+
+        for prepared_statement_index in prepared_statements:
+            prepared_statement = prepared_statements[prepared_statement_index]
+            # TODO (davit) check if this returns anything before fetchall
+            result_proxy = training_dataset.prepared_statement_connection.execute(
+                prepared_statement, entry
+            ).fetchall()
+            for row in result_proxy:
+                result_dict = dict(row.items())
+            serving_vector += list(result_dict.values())
+
+        return serving_vector
+
+    def _init_prepared_statement(self, training_dataset):
+        online_conn = self._storage_connector_api.get_online_connector()
+        jdbc_connection = self._create_mysql_connection(online_conn)
+        prepared_statements = self._training_dataset_api.get_serving_prepared_statement(
+            training_dataset
+        )
+
+        prepared_statements_dict = {}
+        for prepared_statement in prepared_statements:
+            query_online = str(prepared_statement.query_online).replace("\n", "")
+
+            # In java prepared statement `?` is used for prepared statement parametrization.
+            # In sqlalchemy `:feature_name` is used instead of `?`
+            pk_names = [
+                psp.name
+                for psp in sorted(
+                    prepared_statement.prepared_statement_parameters,
+                    key=lambda psp: psp.index,
+                )
+            ]
+            for pk_name in pk_names:
+                query_online = re.sub(
+                    r"^(.*?(\?.*?){0})\?",
+                    r"\1:" + pk_name,
+                    query_online,
+                )
+            query_online = sql.text(query_online)
+
+            prepared_statements_dict[
+                prepared_statement.prepared_statement_index
+            ] = query_online
+
+        training_dataset.prepared_statement_connection = jdbc_connection
+        training_dataset.prepared_statements = prepared_statements_dict
+
+    def _create_mysql_connection(self, online_conn):
```

Comment:
shouldn't this method be in hive-engine? or we have it there already>

Reply:
that's weird, because reading etc. works with it otherwise. make sure it's not an issue of your VM. which `__init__()` throws the error?

Reply:
its Connection ``` __init__()```

```
from sqlalchemy import create_engine

onlconn = fs.get_online_storage_connector()
jdbc_options =  onlconn.spark_options()

sql_alchemy_conn_str = jdbc_options["url"].replace(
            "jdbc:mysql://",
            "mysql+pymysql://"
            + jdbc_options["user"]
            + ":"
            + jdbc_options["password"]
            + "@",
        )
sql_alchemy_engine = create_engine(sql_alchemy_conn_str, pool_recycle=3600)
connection = sql_alchemy_engine.connect()
```
this will throws exception ```got an unexpected keyword argument 'useSSL' and allowPublicKeyRetrieval=true``` . I will try on new VM now

Reply:
you can also try and see if you get the same exception when using hive engine and reading something. just to see if it's a general problem

Reply:
Davit, what's going on here. We really need this PR.

Reply:
Antonios experienced the same issue. Fabio applied the patch here https://github.com/logicalclocks/feature-store-api/commit/6921daf5d36abff05a848073deea7eaa4ed1f030